crates/sui-types/src/transaction.rs

```diff
@@ -239,6 +252,19 @@ impl CallArg {
                     mutable,
                 }]
             }
+            // Receiving objects are not part of the input objects.
+            CallArg::Object(ObjectArg::Receiving(_)) => vec![],
+        }
+    }
+
+    fn receiving_objects(&self) -> Vec<ObjectRef> {
+        match self {
+            CallArg::Pure(_) => vec![],
+            CallArg::Object(o) => match o {
+                ObjectArg::ImmOrOwnedObject(_) => vec![],
+                ObjectArg::SharedObject { .. } => vec![],
+                ObjectArg::Receiving(obj_ref) => vec![*obj_ref],
```

Comment:
Why is this a vec instead of just an option?

Reply:
Don't have a super string opinion here, but I used a vec instead of an option here to keep consistency with the other interfaces around receiving objects, and input objects on PTB/PTB components. In particular both `input_objects` and with other implementations of `receiving_objects` on other parts of the PTB.